### ISSUE TITLE
CI: GitHub Actions

### ISSUE THREAD

**Issue by tueda** [vermaseren/form#361]
[Github Actions](https://docs.github.com/en/actions) offer [free 20 concurrent jobs](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#usage-limits) , while [Travis CI](https://travis-ci.org/) offers [5](https://twitter.com/travisci/status/424990533447327744). Maybe some of the current CI jobs (for example  Valgrind-related jobs), or all of them, could be moved from Google (Travis CI backbone) to Microsoft (GitHub Actions backbone), which gives us a speed gain in CI.

It is also possible to [automatize generating documentation](https://github.com/vermaseren/form/issues/310#issuecomment-558078775) on GitHub.

**Comment by tueda** [comment]
Now, travis-ci.org has been shut down...

**Comment by tueda** [comment]
I have set up GitHub Actions with 2 workflows. `Test` is for building and testing with measuring code coverage to be uploaded to [coveralls.io](https://coveralls.io/github/vermaseren/form). `Deploy` is for building binaries and generating manuals, which will be published to GitHub Releases when the commit has a versioning tag starting with `v`, for example, `v4.2.2` or `v5.0.0`.

Some notes:
- Currently, many tests fail due to #412.
- The `Deploy` workflow contains code to build native Windows binaries but is now disabled because of too many test failures, which need some investigation.
- To install Ubuntu packages, I found https://github.com/awalsh128/cache-apt-pkgs-action very helpful. Unfortunately, it doesn't work with LaTeX packages: https://github.com/awalsh128/cache-apt-pkgs-action/issues/57.
- In the old Travis-CI setup, more tests were available: i386, sanitizers, forcer and clang. Currently, they are not in this GitHub Actions setup.

In the next commit, I will remove the remaining files related to Travis CI and close this issue.
